app/src/androidTest/java/com/bottlerocketstudios/brarchitecture/HomeScreenUITest.kt

```diff
@@ -0,0 +1,115 @@
+package com.bottlerocketstudios.brarchitecture
+
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onRoot
+import com.bottlerocketstudios.brarchitecture.domain.models.GitRepository
+import com.bottlerocketstudios.brarchitecture.domain.models.Link
+import com.bottlerocketstudios.brarchitecture.domain.models.Links
+import com.bottlerocketstudios.brarchitecture.domain.models.User
+import com.bottlerocketstudios.brarchitecture.domain.models.Workspace
+import com.bottlerocketstudios.compose.home.HomeScreen
+import com.bottlerocketstudios.compose.home.HomeScreenState
+import com.bottlerocketstudios.compose.home.UserRepositoryUiModel
+import com.bottlerocketstudios.compose.util.StringIdHelper
+import com.bottlerocketstudios.compose.util.asMutableState
+import com.karumi.shot.ScreenshotTest
+import org.junit.Rule
+import org.junit.Test
+import java.time.ZonedDateTime
+
+class HomeScreenUITest : ScreenshotTest {
+
+    @get:Rule
+    val composeRule = createComposeRule()
+
+    @Test
+    fun homeScreenWithListItemsScreenshot() {
+        renderComponent(
+            HomeScreenState(
+                repositories = listOf(testCard1, testCard2).asMutableState(),
+                itemSelected = {}
+            )
+        )
+
+        compareScreenshot(composeRule.onRoot())
+    }
+
+    @Test
+    fun homeScreenNoResultItemsScreenshot() {
+        renderComponent(
+            HomeScreenState(
+                repositories = emptyList<UserRepositoryUiModel>().asMutableState(),
+                itemSelected = {}
+            )
+        )
+
+        compareScreenshot(composeRule.onRoot())
+    }
+
+    fun renderComponent(state: HomeScreenState? = null) {
```

Comment:
@Br-DiorD I agree with Colin that we should change the type of the state argument to be non nullable as seen below:

```kotlin
fun renderComponent(state: HomeScreenState) {
```